### ISSUE TITLE
Errors encountered while setting up with `requirements.txt`

### ISSUE THREAD

**Issue by zebin-huang** [TheRoboticsClub/gsoc2023-Meiqi_Zhao#2]
#### Issue Summary
I encountered multiple errors while trying to set up a Python environment using `requirements.txt` with both Conda and Pip commands. These issues stem from package version specifications and availability from the channels specified.

#### Environment
- **Operating System:** Ubuntu 20.04.5 LTS
- **Conda Version:** 24.5.0
- **Python Version:** 3.7.16

#### Steps to Reproduce
1. Using Conda with `requirements.txt`:
   - Command: `conda create --name carla_0914_env --file src/requirements.txt`
   - Error Message:

```
Procuda% conda create --name carla_0914_env --file src/requirements.txt

Channels:
 - defaults
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - zipp==3.15.0=pypi_0
  - widgetsnbextension==4.0.8=pypi_0
  - werkzeug==2.2.3=pypi_0
  - wcwidth==0.2.6=pypi_0
  - urllib3==2.0.4=pypi_0
  - typing-extensions==4.7.1=pypi_0
  - traitlets==5.9.0=pypi_0
  - tqdm==4.65.0=pypi_0
  - tornado==6.2=pypi_0
  - torchvision==0.14.1=pypi_0
  - torchaudio==0.13.1=pypi_0
  - torch==1.13.1=pypi_0
  - threadpoolctl==3.1.0=pypi_0
  - tenacity==8.2.2=pypi_0
  - six==1.16.0=pypi_0
  - shapely==2.0.1=pypi_0
  - scipy==1.7.3=pypi_0
  - scikit-learn==1.0.2=pypi_0
  - retrying==1.3.4=pypi_0
  - requests==2.31.0=pypi_0
  - pyzmq==25.1.0=pypi_0
  - pyyaml==6.0.1=pypi_0
  - pytz==2023.3=pypi_0
  - python-dateutil==2.8.2=pypi_0
  - pyrsistent==0.19.3=pypi_0
  - pyquaternion==0.9.9=pypi_0
  - pyparsing==3.1.0=pypi_0
  - pygments==2.15.1=pypi_0
  - pygame==2.5.0=pypi_0
  - ptyprocess==0.7.0=pypi_0
  - psutil==5.9.5=pypi_0
  - prompt-toolkit==3.0.39=pypi_0
  - plotly==5.15.0=pypi_0
  - pkgutil-resolve-name==1.3.10=pypi_0
  - pillow==9.5.0=pypi_0
  - pickleshare==0.7.5=pypi_0
  - pexpect==4.8.0=pypi_0
  - parso==0.8.3=pypi_0
  - pandas==1.3.5=pypi_0
  - packaging==23.1=pypi_0
  - open3d==0.17.0=pypi_0
  - nvidia-cudnn-cu11==8.5.0.96=pypi_0
  - nvidia-cuda-runtime-cu11==11.7.99=pypi_0
  - nvidia-cuda-nvrtc-cu11==11.7.99=pypi_0
  - nvidia-cublas-cu11==11.10.3.66=pypi_0
  - numpy==1.18.4=pypi_0
  - networkx==2.6.3=pypi_0
  - nest-asyncio==1.5.6=pypi_0
  - nbformat==5.7.0=pypi_0
  - matplotlib-inline==0.1.6=pypi_0
  - matplotlib==3.5.3=pypi_0
  - markupsafe==2.1.3=pypi_0
  - kiwisolver==1.4.4=pypi_0
  - jupyterlab-widgets==3.0.8=pypi_0
  - jupyter-core==4.12.0=pypi_0
  - jupyter-client==7.4.9=pypi_0
  - jsonschema==4.17.3=pypi_0
  - joblib==1.3.1=pypi_0
  - jinja2==3.1.2=pypi_0
  - jedi==0.18.2=pypi_0
  - itsdangerous==2.1.2=pypi_0
  - ipywidgets==8.0.7=pypi_0
  - ipython==7.34.0=pypi_0
  - ipykernel==6.16.2=pypi_0
  - importlib-resources==5.12.0=pypi_0
  - importlib-metadata==6.7.0=pypi_0
  - idna==3.4=pypi_0
  - huggingface-hub==0.16.4=pypi_0
  - h5py==3.8.0=pypi_0
  - future==0.18.3=pypi_0
  - fsspec==2023.1.0=pypi_0
  - fonttools==4.38.0=pypi_0
  - flask==2.2.5=pypi_0
  - filelock==3.12.2=pypi_0
  - fastjsonschema==2.18.0=pypi_0
  - entrypoints==0.4=pypi_0
  - decorator==5.1.1=pypi_0
  - debugpy==1.6.7=pypi_0
  - dash-table==5.0.0=pypi_0
  - dash-html-components==2.0.0=pypi_0
  - dash-core-components==2.0.0=pypi_0
  - dash==2.11.1=pypi_0
  - cycler==0.11.0=pypi_0
  - configargparse==1.7=pypi_0
  - click==8.1.6=pypi_0
  - charset-normalizer==3.2.0=pypi_0
  - backcall==0.2.0=pypi_0
  - attrs==23.1.0=pypi_0
  - ansi2html==1.8.0=pypi_0
  - addict==2.4.0=pypi_0

Current channels:

  - defaults

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

 Many packages listed are not available in the default Conda channels, potentially because they are being identified with a `pypi_0` tag.

2. **Using Pip within a virtual environment**
   - Command: `pip install -r src/requirements.txt`
   - Error Message:

 ```
 ERROR: Invalid requirement: '_libgcc_mutex=0.1=main' (from line 4 of gsoc2023-Meiqi_Zhao/src/requirements.txt)
 Hint: = is not a valid operator. Did you mean == ?
 ```

This indicates an incorrect format in the `requirements.txt` file, which is incompatible with Pip's expected syntax.

#### Proposed Solution:
I suggest replacing the existing `requirements.txt` with the `environment.yml` file that works as expected; if needed, I can provide the `environment.yml` file for review or submit a pull request.

**Comment by zebin-huang** [comment]
Hi @sergiopaniego , I've updated the dependencies and installation instructions as suggested #4. Please let me know if there are any further changes needed.

**Comment by sergiopaniego** [comment]
fixed in #4
